dm/checker/checker.go

```diff
@@ -259,7 +264,7 @@ func (c *Checker) Init(ctx context.Context) (err error) {
 					continue
 				}
 				if instance.cfg.ShardMode == config.ShardPessimistic {
-					c.checkList = append(c.checkList, checker.NewShardingTablesChecker(targetTableID, dbs, shardingSet, columnMapping, checkingShardID, dumpThreads))
+					c.checkList = append(c.checkList, checker.NewShardingTablesChecker(targetTableID, dbs, shardingSet, checkingShardID, dumpThreads))
 				} else {
 					c.checkList = append(c.checkList, checker.NewOptimisticShardingTablesChecker(targetTableID, dbs, shardingSet, dumpThreads))
```

Comment:
we should also check AutoID for optimistic sharding mode. Will the optimistic table structure join itself can report the AutoID problem so we don't need extra checking logic?

Reply:
yes, optimistic table structure join can report the autoID problem.

I will add some UT later